File: src/core/cmd/events/system.rs
use serde::{Deserialize, Serialize};

/// System-level events
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(tag = "event", content = "data", rename_all = "kebab-case")]
pub enum SystemEvent {
    /// Application was resumed (from suspended state)
    OnResume,

    /// Application was suspended
    OnSuspend,

    /// Low memory warning
    OnMemoryWarning,

    /// Application is about to exit
    OnExit,
}

File: binding/src/events/system.ts
// MARK: System Events

export interface SystemResumedEvent {
  event: 'on-resume';
}

export interface SystemSuspendedEvent {
  event: 'on-suspend';
}

export interface SystemMemoryWarningEvent {
  event: 'on-memory-warning';
}

export interface SystemExitingEvent {
  event: 'on-exit';
}

export type SystemEvent =
  | SystemResumedEvent
  | SystemSuspendedEvent
  | SystemMemoryWarningEvent
  | SystemExitingEvent;
